src/app/shared/object-collection/shared/listable-object/listable-object-component-loader.component.ts

```diff
@@ -187,7 +187,10 @@ export class ListableObjectComponentLoaderComponent implements OnInit, OnChanges
           this.compRef.destroy();
           this.object = reloadedObject;
           this.instantiateComponent(reloadedObject);
-          this.contentChange.emit(reloadedObject);
+          // Add delay before emitting event to allow the new object is instantiated
+          setTimeout(() => {
```

Comment:
I realized the problem is not to wait for the new object is instantiated on angular side, but is due to the rest because solr indexes should be updated, so a bit delay is necessary before making the new request used to update filters count. I'm aware is not the best solution, i tried to optimize the code using a debounceTime operator instead of setTimeout

Reply:
`debounceTime` is better, but if this is related REST/Solr maybe this should be addressed in REST down the line?

e.g. if task action endpoints could await Solr reindex before responding this delay may no longer be required

Reply:
@ybnd 

I found out the real problem, basically is a cache issue on angular side [here](https://github.com/4Science/dspace-angular/blob/CST-6876/src/app/shared/search/search-filters/search-filter/search-facet-filter/search-facet-filter.component.ts#L284)

What I've observed is the rest request return the response with the updated filter counts, but in the mentioned point I get the old response despite the fact when the filter should be reloaded I pass the `useCachedVersionIfAvailable` [flag to false](https://github.com/4Science/dspace-angular/blob/CST-6876/src/app/shared/search/search-filters/search-filter/search-facet-filter/search-facet-filter.component.ts#L120). For that I was thinking it was an issue on backend side. 
I can't figure it out why with the delay I get the updated response, do you have any suggestion?

Reply:
Looking at this more closely it does seem like there's a lot of nested subscriptions here ([one](https://github.com/DSpace/dspace-angular/pull/1845/files#diff-95b607c8c0460f6ff4f25cdd8806c88892ff1e93da313f4e228d37ac6675e6feR119), [two](https://github.com/DSpace/dspace-angular/pull/1845/files#diff-95b607c8c0460f6ff4f25cdd8806c88892ff1e93da313f4e228d37ac6675e6feR299), [three](https://github.com/DSpace/dspace-angular/pull/1845/files#diff-95b607c8c0460f6ff4f25cdd8806c88892ff1e93da313f4e228d37ac6675e6feR328))

IMO this is the first thing we should look at -- can you refactor this method (and other similar cases) until there's one subscription at most?

Reply:
@ybnd 

I've removed the nested subscriptions, I don't know why with a delay we get the [updated response here](https://github.com/4Science/dspace-angular/blob/CST-6876/src/app/shared/search/search-filters/search-filter/search-facet-filter/search-facet-filter.component.ts#L288)